### DISCUSSION TITLE
I can't seem to include any folders in the archive files

### DISCUSSION THREAD

**Discussion by dogancelik** [goreleaser/goreleaser#1935] [Q&A] [answered]
Hello, after 3 years, after the bot notified me about [this issue](https://github.com/goreleaser/goreleaser/issues/229), I wanted to try Goreleaser again. Downloaded the latest version. (version [`0.149.0`](https://github.com/goreleaser/goreleaser/releases/tag/v0.149.0) commit: a69839327f0015ebb0990d41a6fa73b880de2873)

After trying many combinations, I simply could not, include any folders that I wanted.
* It actually included the `dist/` files inside the Linux tar.gz file (recursively).  
* Then it moved to the Mac archive, and the Mac archive included the Linux archive file as well.
* Then in the Windows archive, the file grew as big as 2 GB then I had to cancel the process.

I even added a random file in `random/file.txt` and added to `files` exactly like that. It didn't even include that.

Here is [the test config I used](https://gist.github.com/dogancelik/d5fd4ea22a03c2cf6972294f3d321b49#file-goreleaser-yml).  
To build the project, you need to checkout the `go-mod` branch [here](https://github.com/bsep/server/tree/go-mod) or you can [download the branch](https://github.com/bsep/server/archive/go-mod.zip).

```yml
    files:
      # main.go works
      - main.go
      # random/file.txt does NOT work
      - random/file.txt
      # **/*, includes dist folder and ignores every other folder, have to quote the string
      - '**/*'
      # ./public/* does NOT work
      - ./public/*
```

The command I use to build and archive the project: `goreleaser release --skip-publish --rm-dist`, it produces [`config.yml`](https://gist.github.com/dogancelik/d5fd4ea22a03c2cf6972294f3d321b49#file-final-config-yml).
Here's [the command output](https://gist.github.com/dogancelik/d5fd4ea22a03c2cf6972294f3d321b49#file-command-output-txt), even though the archives don't include any folders (except `dist`).

Maybe related to #1889?

**Comment by caarlos0** [comment]
You're building on windows right?

I think that might be the issue, we recently changed our globbing library... cc/ @djgilcrease

**Reply by dogancelik** [reply]
Yeah, I'm on Windows 10.

```
Edition	     : Windows 10 Pro
Version	     : 20H2
Installed on : ‎2020.‎10.‎27
OS build     : 19042.662
Experience   : Windows Feature Experience Pack 120.2212.551.0
```

Using the Windows separator (`\`) does not work as well.

```yml
    files:
      - public\*
```

For the sake of testing, I downloaded version [`0.146.0`](https://github.com/goreleaser/goreleaser/releases/tag/v0.146.0) (f629ac74edd09c2c6f59cb1b7473f199bb5b2231),
and yes globbing works with this old glob library version.

**If possible, I want to get the public folder files from another parent-child directory `..\server-web\**\*` and publish it as `public` folder under the dist archive.** Because `server` folder is the Go project, `server-web` is the front-end website of the Go project.

I have seen a notation like [this](https://github.com/cdr/code-server/blob/1826399cd230acc1f62556a5af7bba9c25a03a25/ci/build/nfpm.yaml#L19): `- '../parent_child_folder/**/*' : 'new_folder_name'`

**Reply by caarlos0** [reply]
Can you run `go test -v ./...` on https://github.com/goreleaser/fileglob `main` and `clean` branches and report the results?

**Reply by dogancelik** [reply]
[Here's the test log](https://gist.github.com/dogancelik/d5fd4ea22a03c2cf6972294f3d321b49#file-fileglob-test-log).

I tested `main` first, then `clean` second.

**Reply by caarlos0** [reply]
Yeah, seems like there is a lot of things to fix 😭

**Reply by djgilcrease** [reply]
I have windows so I can take a look at this.

**Reply by caarlos0** [reply]
awesome, thanks @djgilcrease!

We should probably run tests on windows on CI as well... 🤔

**Comment by caarlos0** [comment, marked as answer]
FWIW, due to a go "bug", archiving from windows will not retain the binaries permissions correctly...

that said, not sure if its even worth investigating this further


check #1947
